src/main/java/com/google/devtools/build/lib/remote/ReferenceCountedChannel.java

```diff
@@ -90,21 +88,30 @@ public boolean isShutdown() {
     return dynamicConnectionPool.isClosed();
   }
 
+  /**
+   * A specialized {@link Function} that can only throw {@link IOException} and {@link
```

Comment:
Nit: remove the "only", as makes it sound like `IOFunction` throws a subset of what `Function` throws.